lib/assetmanager.js

```diff
@@ -310,8 +322,9 @@
         var layerIdsWithComponents = [],
             compComponents = [],
             comps = this._document._comps,
-            docMeta = this._getDocumentMetaData();
-        
+            docMeta = this._getDocumentMetaData(),
+            documentAssetSettings = docMeta && docMeta.assetSettings && docMeta.assetSettings[0];
```

Comment:
if this is an array of settings why not just go forEach(this. _addComponentForDocument, this) below? Treat it like an array all the way incase there are more settings at some point